tests/test_draft_resource.py

```diff
@@ -75,59 +116,55 @@ def test_create_draft_of_existing_record(client, record_service,
+    # Publish it to check the increment in reversion
+    response = client.post(
+        "/records/{}/draft/actions/publish".format(recid), headers=HEADERS
+    )
 
-    assert response.json['metadata']['title'] == orig_title
+    assert response.status_code == 202
+    _assert_single_item_response(response)
 
+    assert response.json['metadata']['recid'] == recid
+    assert response.json['revision'] == 1
+    assert response.json['metadata']['title'] == input_record['title']
 
-def test_publish_draft_of_new_record(client, input_record):
-    """Test draft publication of a non-existing record.
+    # Check it was actually edited
+    response = client.get(
+        "/records/{}".format(recid),
+        headers=HEADERS
+    )
+    assert response.json['metadata']['title'] == input_record['title']
 
-    It has to first create said draft.
+
+def test_create_publish_record_new_version(client, input_record,
+                                           fake_identity):
+    """Creates a new version of a record.
+
+    Publishes the draft to obtain 2 versions of a record.
     """
-    # Create the draft
+    recid = _create_and_publish(client, input_record)
+
+    # Create new draft of said record
     response = client.post(
-        "/records", data=json.dumps(input_record), headers=HEADERS
+        "/records/{}/versions".format(recid),
+        headers=HEADERS
     )
 
     assert response.status_code == 201
-    recid = response.json['pid']
+    _assert_single_item_response(response)
+    assert response.json['revision'] == 0
+    recid_2 = response.json['metadata']['recid']
 
-    # Publish it
+    # Publish it to check the increment in reversion
     response = client.post(
-        "/records/{}/draft/actions/publish".format(recid), headers=HEADERS
+        "/records/{}/draft/actions/publish".format(recid_2), headers=HEADERS
     )
 
-    assert response.status_code == 200
-    response_fields = response.json.keys()
-    fields_to_check = ['pid', 'metadata', 'revision',
-                       'created', 'updated', 'links']
-
-    for field in fields_to_check:
-        assert field in response_fields
+    assert response.status_code == 202
+    _assert_single_item_response(response)
 
-    # Check draft deletion
-    # TODO: Remove import when exception is properly handled
-    with pytest.raises(NoResultFound):
-        response = client.get(
-            "/records/{}/draft".format(recid),
-            headers=HEADERS
-        )
-    # assert response.status_code == 404
-
-    # Test record exists
-    response = client.get("/records/{}".format(recid), headers=HEADERS)
-
-    assert response.status_code == 200
-
-    response_fields = response.json.keys()
-    fields_to_check = ['pid', 'metadata', 'revision',
-                       'created', 'updated', 'links']
-
-    for field in fields_to_check:
-        assert field in response_fields
+    assert response.json['metadata']['recid'] == recid_2 != recid
```

Comment:
Nice! For some reason it didn't click for me that a triple equality comparison is possible even if I use `a <= b < c` all the time.

Reply:
TBH, I didn't click on that two. I just said "Oh, would this work?" I checked online, it did... voila! It's nice and might save quite some lines :)